flink-connector/src/main/java/com/google/pubsub/flink/PubSubSource.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.pubsub.flink;
+
+import com.google.api.gax.core.FixedCredentialsProvider;
+import com.google.auth.Credentials;
+import com.google.auto.value.AutoValue;
+import com.google.cloud.pubsub.v1.MessageReceiver;
+import com.google.cloud.pubsub.v1.Subscriber;
+import com.google.pubsub.flink.internal.source.enumerator.PubSubCheckpointSerializer;
+import com.google.pubsub.flink.internal.source.enumerator.PubSubSplitEnumerator;
+import com.google.pubsub.flink.internal.source.reader.AckTracker;
+import com.google.pubsub.flink.internal.source.reader.PubSubNotifyingPullSubscriber;
+import com.google.pubsub.flink.internal.source.reader.PubSubRecordEmitter;
+import com.google.pubsub.flink.internal.source.reader.PubSubSourceReader;
+import com.google.pubsub.flink.internal.source.reader.PubSubSplitReader;
+import com.google.pubsub.flink.internal.source.split.SubscriptionSplit;
+import com.google.pubsub.flink.internal.source.split.SubscriptionSplitSerializer;
+import com.google.pubsub.flink.proto.PubSubEnumeratorCheckpoint;
+import com.google.pubsub.v1.ProjectSubscriptionName;
+import java.util.HashMap;
+import java.util.Optional;
+import org.apache.flink.api.common.serialization.DeserializationSchema;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.connector.source.Boundedness;
+import org.apache.flink.api.connector.source.Source;
+import org.apache.flink.api.connector.source.SourceReader;
+import org.apache.flink.api.connector.source.SourceReaderContext;
+import org.apache.flink.api.connector.source.SplitEnumerator;
+import org.apache.flink.api.connector.source.SplitEnumeratorContext;
+import org.apache.flink.api.java.typeutils.ResultTypeQueryable;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.core.io.SimpleVersionedSerializer;
+import org.apache.flink.metrics.MetricGroup;
+import org.apache.flink.util.UserCodeClassLoader;
+
+@AutoValue
+public abstract class PubSubSource<OutputT>
```

Comment:
Please split the source from its setting object, like PubSubLiteSource

Reply:
My preference would be to keep this as is because the plan is to migrate this connector to  https://github.com/apache/flink-connector-gcp-pubsub to replace the apache owned connector.

Having the builder apart of Source object is more consistent with the existing connector (and other existing connectors in the apache repo).